src/box/tuple.h

```diff
@@ -356,9 +357,55 @@ enum {
 	TUPLE_REF_MAX = UINT8_MAX,
 };
 
+/**
+ * Check that data_offset is valid and can be stored in tuple.
+ * @param data_offset - see member description in struct tuple.
+ *
+ * @return 0 on success, -1 otherwise (diag is set).
+ */
+static inline int
+tuple_check_data_offset(uint32_t data_offset)
+{
+	if (data_offset > INT16_MAX) {
+		/** tuple->data_offset is 15 bits */
+		diag_set(ClientError, ER_TUPLE_METADATA_IS_TOO_BIG,
+			 data_offset);
+		return -1;
+	}
+	return 0;
+}
+
+/**
+ * Initialize a tuple. Must be called right after allocation of a new tuple.
+ * Should only be called for newly created uninitialized tuples.
+ * If the tuple copied from another tuple and only initialization of reference
+ * count is needed, it's better to call tuple_ref_init.
+ * tuple_check_data_offset should be called before to ensure args are correct.
+ *
+ * @param tuple - Tuple to initialize.
+ * @param refs - initial reference count to set.
+ * @param format_id - see member description in struct tuple.
+ * @param data_offset - see member description in struct tuple.
+ * @param bsize - see member description in struct tuple.
+ * @return 0 on success, -1 on error (diag is set).
+ */
+static inline void
+tuple_init_default(struct tuple *tuple, uint8_t refs, uint16_t format_id,
```

Comment:
We typically use a function named 'something_init' to init a module, not an object. We call functions that initialize an object 'something_create'.

Also, _default is not quite an appropriate suffix IMO. What about _generic? tuple_create_generic?

Reply:
Will simple `tuple_create` be OK? I'll try.

Reply:
Sounds good to me.